### ISSUE TITLE
[BUG] MacOS downloading Issue

### ISSUE THREAD

**Issue by Hossra** [Plompi/VinURL#22]
**Describe the bug**
"Failed to download music!" message after waiting for music URL to download.

**To Reproduce**
1. Right click Custom Music Disc and paste youtube URL. 
(/vinurl set command gives "Expected whitespace to end one argument, but found trailing data at position 16..."
2. Receive "Downloading music, please wait a moment" message
5. PC players have music finish downloading and begin to play
6. Mac player gets "Failed to download music!" message

**Expected behavior**
All players at jukebox should be able to download the song and listen to it.

**Screenshots**
![2024-09-22_18 13 54](https://github.com/user-attachments/assets/5d742740-2a3e-4861-9e0a-209092346470)

**Desktop (please complete the following information):**
 - OS: MacOS Monterey 
 - MC Version 1.20.1
 - Mod Version 1.1.2

**Additional context**
Again, this mod works perfectly for the two players who are on PC. Not sure if this is an OS related issue, but it is the only variable we can see between the one player unable to download music and the other two.

**Comment by Plompi** [comment]
thank you very much for the bug report, it looks like the problem is operating system dependent. Unfortunately I don't have a computer with MacOS and can therefore theoretically develop the mod for MacOS, but cannot test it. I could imagine that this is a problem with permissions on MacOS.

**Comment by nichols7** [comment]
I encountered this issue today after installing VinURL (excellent job with the mod btw!). Found this issue marked open and went to work figuring out a solution. In my instance (docker container multiplayer server on debian bookworm with a client windows PC working as expected), I experienced the same symptoms as Hossra on my macbook, and the latest logs were of little help most likely due to my inexperience with java but I localized these log entries to right after I inserted a disc each time:

> [15:53:54] [ForkJoinPool.commonPool-worker-1/WARN]: ERROR: expected str, bytes or os.PathLike object, not NoneType
> [15:53:54] [ForkJoinPool.commonPool-worker-1/ERROR]: Command failed with exit code: 1

I tried a few permissions changes, however it seemed like everything that needed to be executable was in fact executable per `ls -al`

I attempted to open the auto-downloaded ffmpeg & ffprobe and I was notified by each in the terminal that they needed a newer version of macOS and would exit with an error. Next, I downloaded the latest macOS 64-bit binaries from [ffmpeg.org] and replaced both ffmpeg and ffprobe in the vinURL folder. Additionally I made sure that permissions for both were set to 755, -rwxr-xr-x via `chmod` The mod is successfully working on my 2017 MBP running macOS Ventura. 

Of note: in case a previous attempted route of mine was _also_ necessary for success--I had installed yt-dlp locally on my macOS machine via brew earlier which didn't resolve anything.

Also Plompi, thank you for this awesome mod!! I only included any elementary coding details in case someone without the requisite knowledge decides to follow my write-up prior to a resolution to this issue.

**Comment by Plompi** [comment]
> I encountered this issue today after installing VinURL (excellent job with the mod btw!). Found this issue marked open and went to work figuring out a solution. In my instance (docker container multiplayer server on debian bookworm with a client windows PC working as expected), I experienced the same symptoms as Hossra on my macbook, and the latest logs were of little help most likely due to my inexperience with java but I localized these log entries to right after I inserted a disc each time:
> 
> > [15:53:54] [ForkJoinPool.commonPool-worker-1/WARN]: ERROR: expected str, bytes or os.PathLike object, not NoneType
> > [15:53:54] [ForkJoinPool.commonPool-worker-1/ERROR]: Command failed with exit code: 1
> 
> I tried a few permissions changes, however it seemed like everything that needed to be executable was in fact executable per `ls -al`
> 
> I attempted to open the auto-downloaded ffmpeg & ffprobe and I was notified by each in the terminal that they needed a newer version of macOS and would exit with an error. Next, I downloaded the latest macOS 64-bit binaries from [ffmpeg.org] and replaced both ffmpeg and ffprobe in the vinURL folder. Additionally I made sure that permissions for both were set to 755, -rwxr-xr-x via `chmod` The mod is now successfully working as intended on my 2017 MBP running macOS Ventura.
> 
> Of note: in case a previous attempted route of mine was _also_ necessary for success--I had installed yt-dlp locally on my macOS machine via brew earlier which didn't resolve anything.
> 
> Also Plompi, thank you for this awesome mod!! I only included any elementary coding details in case someone without the requisite knowledge decides to follow my write-up prior to a resolution to this issue.

Wow, thank you very much for the deep analysis. Could you try out the binaries from this (repo)[https://github.com/Tyrrrz/FFmpegBin/releases ] ? Especially the ones with osx in the name. Currently the mod downloads the x64 macos one, maybe thats the issue cause newer macbooks are ARM based.

**Comment by nichols7** [comment]
> Wow, thank you very much for the deep analysis. Could you try out the binaries from this [repo](https://github.com/Tyrrrz/FFmpegBin/releases) ? Especially the ones with osx in the name. Currently the mod downloads the x64 macos one, maybe thats the issue cause newer macbooks are ARM based.


I ran the x64 binaries from the repo you sent and they both produced the following abort message in my terminal (identifying strings redacted):

> /Users/**user**/Downloads/ffmpeg-osx-x64/ffmpeg ; exit;
dyld[71525]: Symbol not found: _AVCaptureDeviceTypeContinuityCamera
  Referenced from: <***-***-***-***-**1> /Users/**user**/Downloads/ffmpeg-osx-x64/ffmpeg (built for macOS 14.0 which is newer than running OS)
  Expected in:     <***-***-***-***-**2> /System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation
[1]    71525 abort      /Users/**user**/Downloads/ffmpeg-osx-x64/ffmpeg

Saving session...completed.


The repo is providing binaries for macOS 14 (Sonoma) which is an OS for both x86_64 and ARM CPU architecture, however the binary does not support operating systems prior to macOS 14. The binaries that I have found to work are the most recently compiled macOS x64 binaries from ffmpeg's official site, however the site links to download at [https://evermeet.cx/ffmpeg/](url). I believe these binaries support all x64 mac operating systems since 10.9 which should cover almost anyone playing MC on an intel mac today. The maintainer at evermeet does not have the binaries stored in a github repo and I was unable to locate one. I think that if you were to upload the latest binary snapshots of ffmpeg and ffprobe from that link into your repo that should just about cover all intel macs for good, as they haven't been sold with x64 chips since the year 2020.

As an aside... I love all 3 major operating systems and I do know these kind of issues would be alleviated if I used either windows or linux, however I prefer mac for laptop work (and mild gaming away from home), windows for gaming desktop, and linux for server. Hopefully this will help out the folks that either won't stray from mac or have the same preferences as I do and just want to play a little MC with their kids when they're away!!

Thanks again Plompi :)

**Comment by Plompi** [comment]
> > Wow, thank you very much for the deep analysis. Could you try out the binaries from this [repo](https://github.com/Tyrrrz/FFmpegBin/releases) ? Especially the ones with osx in the name. Currently the mod downloads the x64 macos one, maybe thats the issue cause newer macbooks are ARM based.
> 
> I ran the x64 binaries from the repo you sent and they both produced the following abort message in my terminal (identifying strings redacted):
> 
> > /Users/**user**/Downloads/ffmpeg-osx-x64/ffmpeg ; exit;
> > dyld[71525]: Symbol not found: _AVCaptureDeviceTypeContinuityCamera
> > Referenced from: <_**-**_-_**-**_-**1> /Users/user/Downloads/ffmpeg-osx-x64/ffmpeg (built for macOS 14.0 which is newer than running OS)
> > Expected in:     <**_-_**-**_-_**-**2> /System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation
> > [1]    71525 abort      /Users/**user**/Downloads/ffmpeg-osx-x64/ffmpeg
> > Saving session...completed.
> 
> The repo is providing binaries for macOS 14 (Sonoma) which is an OS for both x86_64 and ARM CPU architecture, however the binary does not support operating systems prior to macOS 14. The binaries that I have found to work are the most recently compiled macOS x64 binaries from ffmpeg's official site, however the site links to download at [https://evermeet.cx/ffmpeg/](url). I believe these binaries support all x64 mac operating systems since 10.9 which should cover almost anyone playing MC on an intel mac today. The maintainer at evermeet does not have the binaries stored in a github repo and I was unable to locate one. I think that if you were to upload the latest binary snapshots of ffmpeg and ffprobe from that link into your repo that should just about cover all intel macs for good, as they haven't been sold with x64 chips since the year 2020.
> 
> As an aside... I love all 3 major operating systems and I do know these kind of issues would be alleviated if I used either windows or linux, however I prefer mac for laptop work (and mild gaming away from home), windows for gaming desktop, and linux for server. Hopefully this will help out the folks that either won't stray from mac or have the same preferences as I do and just want to play a little MC with their kids when they're away!!
> 
> Thanks again Plompi :)

could you try the newly released 1.7.1? I changed the repository for ffmpeg/ffprobe the the one mentioned above

**Comment by nichols7** [comment]
Absolutely! I will run tests early this week, I have a busy next couple of days. I'll get back to you as soon as I can!

Thanks again for all of your hard work,

Pete
